services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/grpc/LogicalPortSpec.groovy

```diff
@@ -0,0 +1,98 @@
+package org.openkilda.functionaltests.spec.grpc
+
+import org.openkilda.grpc.speaker.model.LogicalPortDto
+import org.openkilda.messaging.error.MessageError
+
+import groovy.util.logging.Slf4j
+import org.springframework.http.HttpStatus
+import org.springframework.web.client.HttpClientErrorException
+import spock.lang.Narrative
+import spock.lang.Shared
+import spock.lang.Unroll
+
+@Slf4j
+@Narrative("""This test suite checks the CRUD actions on a logical port.
+Logical ports are defined by associating a single physical port to them to define
+Bidirectional Forwarding Detection(BFD) ports or
+by associating a list of physical ports to them to create Link Aggregation Groups (LAG) or
+a list of BFD ports to them to create a LAG for fast-failover for BFD sessions.
+
+NOTE: The GRPC implementation supports the LAG type only and it is set by default.""")
+class LogicalPortSpec extends GrpcBaseSpecification {
+    @Shared
+    Integer switchPort
+    @Shared
+    Integer switchLogicalPort
+
+    def setup() {
+        switchPort = northbound.getPorts(nFlowSwitch.switchId).find { it.state[0] == "LINK_DOWN" }.portNumber
+        switchLogicalPort = 1100 + switchPort
+    }
+
+    def "Able to create/read/delete logicalport"() {
+        /**the update action is not working(issue on a Noviflow switch side)*/
+        when: "Create logical port"
+        grpc.createLogicalPort(switchIp, new LogicalPortDto([switchPort], switchLogicalPort))
+
+        then: "Able to get the created logical port"
+        def r = grpc.getSwitchLogicalPortConfig(switchIp, switchLogicalPort)
```

Comment:
please avoid such variable names. What is `r` ?

Reply:
fixed